src/function/table/arrow_conversion.cpp

```diff
@@ -294,6 +294,35 @@ static void SetVectorString(Vector &vector, idx_t size, char *cdata, T *offsets)
 	}
 }
 
+static void SetVectorStringView(Vector &vector, idx_t size, ArrowArray &array) {
+	auto strings = FlatVector::GetData<string_t>(vector);
+	auto length_buffer = ArrowBufferData<int32_t>(array, 1);
+	auto views_buffer = ArrowBufferData<char>(array, 1);
+
+	for (idx_t row_idx = 0; row_idx < size; row_idx++) {
+		if (FlatVector::IsNull(vector, row_idx)) {
+			continue;
+		}
+		int32_t length = length_buffer[row_idx * 4];
+		if (length <= 12) {
+			//	This string is inlined
+			//  | Bytes 0-3  | Bytes 4-15                            |
+			//  |------------|---------------------------------------|
+			//  | length     | data (padded with 0)                  |
```

Comment:
Zero padding is necessary for the whole word comparison shortcuts: when sorting, instead of comparing inline strings byte by byte we can reinterpret them as integers and compare in one instruction. If the padding isn't zeroed, the integers don't compare consistently